ddl/index.go

```diff
@@ -495,7 +498,7 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 	case model.StateNone:
 		// none -> delete only
 		indexInfo.State = model.StateDeleteOnly
-		updateHiddenColumns(tblInfo, indexInfo, model.StateDeleteOnly)
+		updateHiddenColumns(tblInfo, indexInfo, model.StatePublic)
```

Comment:
```
mysql> create table t(a int default 0, b int default 0, _V$_idx_0 int default 1);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t(a, b) values (1, 1), (2, 2), (3, 3), (4, 4);
Query OK, 4 rows affected (0.01 sec)
Records: 4  Duplicates: 0  Warnings: 0

mysql> alter table t add index idx((b+1));
ERROR 1060 (42S21): Duplicate column name '_V$_idx_0'
```

master still has this problem, maybe we can ajust the expression index column dynamically.

Reply:
It's exptected